src/Common/Method/ExistsPostfixRule.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace ArchitectureSniffer\Common\Method;
+
+use PHPMD\AbstractNode;
+use PHPMD\AbstractRule;
+use PHPMD\Node\ClassNode;
+use PHPMD\Node\MethodNode;
+use PHPMD\Rule\ClassAware;
+
+class ExistsPostfixRule extends AbstractRule implements ClassAware
+{
+    /**
+     * @var string
+     */
+    protected const POSTFIX_PATTERN = '/Exists?([A-Z]|$)/';
+
+    /**
+     * @var string
+     */
+    protected const POSTFIX_EXISTS = 'Exists';
+
+    /**
+     * @var string
+     */
+    protected const POSTFIX_EXIST = 'Exist';
+
+    /**
+     * @var string
+     */
+    protected const POSTFIX_EXISTING = 'Existing';
+
+    /**
+     * @var string
+     */
+    protected const PREFIX_METHOD_NAME = 'is';
+
+    /**
+     * @var string
+     */
+    protected const RULE_EXISTS_INSTEAD_OF_EXIST = 'Postfix `Exists` must be used instead of `Exist`. Method: %s.';
```

Comment:
Just a side note: Messages should usually not be part of class const.
No blocker for now, just for future
It makes it less readable and is an anti pattern as those are never needed to be reused or extended.
They are easier to use in the place where they are needed

Same for other non reusable/extendable values